### PR TITLE
Speedups

### LINKED ISSUE #8: Broken on LineString
Just noticed this viewing the code: https://github.com/Turfjs/turf-line-distance/blob/master/index.js#L39

### PR DESCRIPTION
**Pull request by calvinmetcalf**
fixes #8 and #9

### CODE REVIEW

**Comment by mourner** [comment]
Not entirely. 2 times less point creation is nice, but we need to fully eradicate the need to use `turf.point` here. Perhaps this will require hacking on https://github.com/Turfjs/turf-distance too.

**Comment by calvinmetcalf** [comment]
well this brings it down to only having to create 2 objects, though that is 2 too many

File: index.js [hunk 1]
```diff
@@ -36,12 +35,32 @@ var point = require('turf-point');
...
+      coordinates: coords[0]
+    }
+  };
+  var curCoords = {
+      type: 'Feature',
+      properties: {},
+      geometry: {
+        type: 'Point'
+      }
+    };
```
Comment (mourner) [inline index.js]: we can cut some lines here with `var prevCoords = point(coords[0])`, same for `curCoords`.

**Comment by mourner** [comment]
@calvinmetcalf ha, that was smart!
I still think this should be upstream (turf-distance should accept points as `[x, y]` in addition to feature or geometry), but the fix is cool for now. :)

**Comment by morganherlocker** [comment]
> I still think this should be upstream (turf-distance should accept points as [x, y] in addition to feature or geometry)

I am pretty strongly against this, since geojson is _the_ data interop format for turf. I think the best option would be another library that accepts raw coords, then turf-distance and turf-line-distance would both depend on that.

**Comment by mourner** [comment]
>  I think the best option would be another library that accepts raw coords, then turf-distance and turf-line-distance would both depend on that.

Yeah, that sounds good. The problem is that `turf.distance` is used in hot functions as well... But maybe a note in the docs pointing to the newly created distance module for raw coords would enough.

File: index.js (hunk 1 above)
Comment (calvinmetcalf) [inline index.js]: done

**Comment by calvinmetcalf** [comment]
so the other option is to use something like Turfjs/turf-distance#13 which makes the internal functions available so this library could calculate R once instead of every time and avoid all the checks and object creation.

**Comment by calvinmetcalf** [comment]
current

``` bash
maboscmetcalf-m1:turf-line-distance cmetcalf$ node bench.js
turf-line-distance#simple x 878,364 ops/sec ±1.25% (94 runs sampled)
turf-line-distance#200 mi x 935 ops/sec ±1.30% (94 runs sampled)
turf-line-distance#700 km x 1,109 ops/sec ±0.93% (94 runs sampled)
maboscmetcalf-m1:turf-line-distance cmetcalf$ node bench.js
turf-line-distance#simple x 873,897 ops/sec ±0.91% (92 runs sampled)
turf-line-distance#200 mi x 935 ops/sec ±1.33% (94 runs sampled)
turf-line-distance#700 km x 1,107 ops/sec ±0.84% (93 runs sampled)
```

this pull

```
maboscmetcalf-m1:turf-line-distance cmetcalf$ node bench.js
turf-line-distance#simple x 964,946 ops/sec ±0.93% (88 runs sampled)
turf-line-distance#200 mi x 1,047 ops/sec ±1.43% (91 runs sampled)
turf-line-distance#700 km x 1,244 ops/sec ±1.08% (91 runs sampled)
maboscmetcalf-m1:turf-line-distance cmetcalf$ node bench.js
turf-line-distance#simple x 981,277 ops/sec ±1.45% (92 runs sampled)
turf-line-distance#200 mi x 1,071 ops/sec ±1.28% (93 runs sampled)
turf-line-distance#700 km x 1,273 ops/sec ±0.86% (95 runs sampled)
```

punching hole in it

```
maboscmetcalf-m1:turf-line-distance cmetcalf$ node bench.js
turf-line-distance#simple x 1,154,872 ops/sec ±0.95% (95 runs sampled)
turf-line-distance#200 mi x 1,253 ops/sec ±1.88% (93 runs sampled)
turf-line-distance#700 km x 1,490 ops/sec ±1.21% (95 runs sampled)
maboscmetcalf-m1:turf-line-distance cmetcalf$ node bench.js
turf-line-distance#simple x 1,137,542 ops/sec ±1.24% (93 runs sampled)
turf-line-distance#200 mi x 1,260 ops/sec ±1.10% (93 runs sampled)
turf-line-distance#700 km x 1,468 ops/sec ±1.71% (92 runs sampled)
```

**Comment by mourner** [comment]
Hmm, not as huge of a speed up as I expected, maybe there's something else going on in `turf.distance`. Still nice though. I like the option of exposing internal distance function — better than a separate module imho.

**Comment by mourner** [comment]
Hmm, benchmarks don't reflect performance scaling well — it's run on 6-point line. How about something like a 1,000-point line?

**Comment by calvinmetcalf** [comment]
the 200 mile test is like 4k points

**Comment by mourner** [comment]
Ah right, didn't notice. Weird.

**Comment by mourner** [comment]
Looked at turf-distance a bit. It looks like the speed up is not as substantial as I hoped because trigonometric functions take a lot of time so the object creation overhead isn't as significant in comparison — not much we can do here further. I made a commit to make it 15-20% faster though.

And the 30% speed up of this PR (the third way) is still very substantial.

**Pull request merged**